### ISSUE TITLE
FastAPI showing the requests on datadog but not extra tags

### ISSUE THREAD

**Issue by ZdsAlpha** [DataDog/dd-trace-py#3614]
Thanks for taking the time for reporting an issue!

Before reporting an issue on dd-trace-py, please be sure to provide all
necessary information.

If you're hitting a bug, make sure that you're using the latest version of this
library.

### Which version of dd-trace-py are you using?

Latest, just installed.

_ddtrace requires pip>=18 to install one of our pre-built wheels_

pip 21.2.4

You can copy/paste the output of `pip freeze` here.

absl-py==1.0.0
addict==2.4.0
aiohttp==3.8.1
aiosignal==1.2.0
albumentations==1.1.0
antlr4-python3-runtime==4.8
anyio==3.5.0
appdirs==1.4.4
asgiref==3.5.0
async-generator==1.10
async-timeout==4.0.1
attrs==21.2.0
black==21.4b2
Brotli==1.0.9
cachetools==4.2.4
certifi==2021.10.8
cffi==1.15.0
charset-normalizer==2.0.7
click==7.1.2
cloudpickle==2.0.0
colorama==0.4.4
ConfigArgParse==1.5.3
cryptography==36.0.1
cycler==0.11.0
Cython==0.29.24
ddtrace==1.0.1
detectron2 @ git+https://github.com/facebookresearch/detectron2.git@3fd63b217e028b4cce2a052e765f136d1dea255b
easyocr==1.4.1
fastapi==0.73.0
Flask==2.0.2
Flask-BasicAuth==0.2.0
Flask-Cors==3.0.10
fonttools==4.28.3
frozenlist==1.2.0
fsspec==2021.11.1
future==0.18.2
fvcore==0.1.5.post20211023
gevent==21.12.0
geventhttpclient==1.5.3
google-auth==2.3.3
google-auth-oauthlib==0.4.6
greenlet==1.1.2
grpcio==1.41.1
h11==0.13.0
hydra-core==1.1.1
idna==3.3
imageio==2.9.0
iopath==0.1.9
itsdangerous==2.0.1
jellyfish==0.8.9
Jinja2==3.0.3
joblib==1.1.0
kiwisolver==1.3.2
llvmlite==0.37.0
locust==2.7.3
Markdown==3.3.4
MarkupSafe==2.0.1
matplotlib==3.4.3
mkl-fft==1.3.1
mkl-random @ file:///tmp/build/80754af9/mkl_random_1626186066731/work
mkl-service==2.4.0
mmcv-full==1.4.2
mmdet @ file:///tmp/tmp4yworwzq/mmdetection
model-index==0.1.11
msgpack==1.0.3
multidict==5.2.0
mypy-extensions==0.4.3
networkx==2.6.3
numba==0.54.1
numpy==1.20.3
oauthlib==3.1.1
olefile @ file:///Users/ktietz/demo/mc3/conda-bld/olefile_1629805411829/work
omegaconf==2.1.1
opencv-python==4.5.5.62
opencv-python-headless==4.5.4.58
openmim==0.1.5
ordered-set==4.0.2
outcome==1.1.0
packaging==21.3
pandas==1.3.5
pathspec==0.9.0
Pillow==8.2.0
portalocker==2.3.2
protobuf==3.19.1
psutil==5.9.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycocotools==2.0.2
pycparser==2.21
pydantic==1.9.0
pyDeprecate==0.3.1
pydot==1.4.2
pyOpenSSL==21.0.0
pyparsing==3.0.6
python-bidi==0.4.2
python-dateutil==2.8.2
python-multipart==0.0.5
pytorch-lightning==1.5.5
pytz==2021.3
PyWavelets==1.2.0
PyYAML==6.0
pyzmq==22.3.0
qudida==0.0.4
regex==2021.11.10
requests==2.26.0
requests-oauthlib==1.3.0
roundrobin==0.0.2
rsa==4.7.2
scikit-image==0.19.1
scikit-learn==1.0.2
scipy==1.7.3
selenium==4.1.0
Shapely==1.8.0
six @ file:///tmp/build/80754af9/six_1623709665295/work
sniffio==1.2.0
sortedcontainers==2.4.0
starlette==0.17.1
tabulate==0.8.9
tenacity==8.0.1
tensorboard==2.7.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.0
termcolor==1.1.0
terminaltables==3.1.10
threadpoolctl==3.0.0
tifffile==2021.11.2
toml==0.10.2
torch==1.10.0
torchaudio==0.10.0
torchmetrics==0.6.1
torchvision==0.11.1
tqdm==4.62.3
trio==0.19.0
trio-websocket==0.9.2
typing-extensions @ file:///tmp/build/80754af9/typing_extensions_1631814937681/work
urllib3==1.26.7
uvicorn==0.17.4
Werkzeug==2.0.2
wsproto==1.0.0
yacs==0.1.8
yapf==0.32.0
yarl==1.7.2
zope.event==4.5.0
zope.interface==5.4.0

### How can we reproduce your problem?
Set-up env and local dd service and set-up DD_SERVICE and DD_ENV variables.

```
@app.post(
    "/predict",
    response_model=List[SimpleResult],
    dependencies=[Depends(auther)],
)
@tracer.wrap("predict", "fastapi")
def predict(
    image: bytes = File(...),
    format: str = "simple",
):
    # do stuff
    span = tracer.current_span()
    assert span
    if span:
        h, w = image.shape[:2]
        span.set_tag("image.height", str(h))
        span.set_tag("image.width", str(w))
    return result
```

### What is the result that you get?

![image](https://user-images.githubusercontent.com/13795176/164994487-5ffd812d-1e32-4094-b863-e83ea775e2a4.png)

### What is the result that you expected?

Output to have image.height and image.width attributes.

**Comment by brettlangdon** [comment]
@ZdsAlpha thank you for opening this issue. Without seeing the trace/other information it appears you are setting the tags on the `@tracer.wrap("predict", "fastapi")` span, but looking for the tags on the root `fastapi.request` span?

If your goal is to have those tags added to the root span, then you will want to do:

```python
    span = tracer.current_root_span()  # Get the root `fastapi.request` span
    assert span
    if span:
        h, w = image.shape[:2]
        span.set_tag("image.height", str(h))
        span.set_tag("image.width", str(w))
```

**Comment by ZdsAlpha** [comment]
I updated the code and still got same results:

```
    span = tracer.current_root_span()
    assert span
    if span:
        h, w = image.shape[:2]
        span.set_tag("image.height", str(h))
        span.set_tag("image.width", str(w))
```
![image](https://user-images.githubusercontent.com/13795176/165186309-2adf0e93-b944-4786-bfa2-967d581c8724.png)

**Comment by brettlangdon** [comment]
🤔 

@ZdsAlpha would you be willing to open a case with support@datadoghq.com, it might be easier for us to look at some of your traces / dig into your setup further.

**Comment by ZdsAlpha** [comment]
Solved the issue by manually patching fastapis instead of running with dd-trace command.

**Comment by kgadek93** [comment]
Are there any updates? I'm still having this issue.
@ZdsAlpha could you post how you've fixed this locally?

**Comment by ZdsAlpha** [comment]
Sorry for late response. You need to manually patch fastapi in python script.

`https://ddtrace.readthedocs.io/en/stable/integrations.html`
